invenio_records_permissions/generators.py

```diff
@@ -103,26 +103,54 @@ def query_filter(self, record=None, **kwargs):
 
 
 class AnyUserIfPublic(Generator):
-    """Allows any user if record is public.
+    """Allows any user if record is open access.
 
     TODO: Revisit when dealing with files.
     """
 
     def needs(self, record=None, **rest_over):
         """Enabling Needs."""
-        is_restricted = (
-            record and
-            record.get('_access', {}).get('metadata_restricted', False)
+        is_open = (
+            record and record.get('access_right') == "open"
         )
-        return [any_user] if not is_restricted else []
+        return [any_user] if is_open else []
 
-    def excludes(self, record=None, **rest_over):
-        """Preventing Needs."""
-        return []
+    def query_filter(self, *args, **kwargs):
+        """Filters for non-restricted records."""
+        return Q('term', **{"access_right": "open"})
+
+
+class AllowedIdentities(Generator):
+    """Allows explicit users."""
+
+    def __init__(self, action='read'):
+        """Constructor."""
+        self.action = action
+        self.can = 'can_' + str(self.action)
+
+    def needs(self, record=None, **kwargs):
+        """Enabling UserNeeds for each person.
+
+        TODO: Organization and role needs
+        """
+        if not record:
+            return []
+
+        return [
+            UserNeed(identity.get('id')) for identity in
```

Comment:
Medium: `Identity` could be mistaken with the global flask principal one  (In fact I got it mistaken). Is there any other possible naming?

Reply:
I see what you mean and too be honest that was my goal: to make this "identity" be the equivalent/correlate to the flask-principal identity while not being one itself. It is these "kinds" of identities that can be targeted by permissions.

The other names I had in mind are: `permission_identity` (too similar perhaps), `permission_actor`, `permission_agent`. Which sounds good to you?